### ISSUE TITLE
1.18 support

### ISSUE THREAD

**Issue by TrinTragula** [Cubitect/cubiomes#57]
Are there any plans on supporting 1.18, sooner or later?
I built a web app ([mcseeder.com](https://www.mcseeder.com), the code is open source) around your project and I would love to be able to update it to the new version once it comes out. I'll even try to give you some help, if needed, but I would be almost new to programming in C so I wouldn't be that much helpful, I guess!

**Comment by Cubitect** [comment]
I have been working on 1.18 support for a while now in the back, and I have just pushed the update.
The API has changed though, to accommodate the different Minecraft versions and 3D biome generation. I will probably update the readme in the coming days, which will probably need a rewrite. Also the structure generation for 1.18 is not well tested jet.

**Comment by TrinTragula** [comment]
I'll try to reverse engineer the code while waiting for an updated readme, then! Thanks as always for the incredible work.

**Comment by TrinTragula** [comment]
One quick question: yesterday night I did a rough implementation of the features I needed but it seems like the `checkForBiomes` function is still a WIP, right? Also, if you're looking for some feedback I must say that the new APIs as of now look great, much easier to use and even more powerful

**Comment by TrinTragula** [comment]
Another thing i noticed, I think the scale factors of 1.18 are doubled or different with respect to the ones of the other versions.

**Comment by TrinTragula** [comment]
Nevermind, it actually looks like they radically changed everything, the scale is fine!

**Comment by TrinTragula** [comment]
Found a possible bug: the world spawn for older MC versions as calculated with the latest cubiomes version is different from the spawn calculated with older cubiomes versions for the same seed, and the spawn location for 1.18 seems to usually be wrong

**Comment by Cubitect** [comment]
I introduced a bug while preparing the update for the spawn finder. They should be ready now, and the spawn and stronghold finders should hopefully work (again).
Unfortunately the strongholds in particular are highly dependent on the order of generation and I had to emulate [MC-241546](https://bugs.mojang.com/browse/MC-241546) to try and get most of the positions right. I really hope Mojang fixes that bug before the release, as stronghold positions are going to be completely different when they do. But this late in the snapshots I fear they are not going to.
